### ISSUE TITLE
Reindexing job for the (unpublished) Stanford Band exhibit is hung

### ISSUE THREAD

**Issue by caaster** [sul-dlss/exhibits#350]
Please clear hanging reindex job so that it can be re-initiated by Daniel Hartwig.
<img width="917" alt="screen shot 2016-08-31 at 9 32 20 am" src="https://cloud.githubusercontent.com/assets/3210225/18137608/df3e6e14-6f5d-11e6-8f27-d090a3e912de.png">

**Comment by caaster** [comment]
What seems strange is that I pushed the reindex button for this job at 1:53 pm today. The job immediately reflected the status/numbers in the attached screenshot, the numbers haven't budged since doing so, and funny enough the numbers & status bar in this screenshot look exactly like the screen shot I captured when I filed the ticket originally.
![screen shot 2016-08-31 at 2 16 54 pm](https://cloud.githubusercontent.com/assets/3210225/18146705/05a3ed5e-6f86-11e6-8373-aa79ae63d2fc.png)

**Comment by caaster** [comment]
Reindexing jobs are still not completing.
<img width="945" alt="screen shot 2016-09-02 at 1 27 06 pm" src="https://cloud.githubusercontent.com/assets/3210225/18217332/15269f9c-7111-11e6-9a1f-595fb2052eab.png">
<img width="1020" alt="screen shot 2016-09-02 at 1 26 19 pm" src="https://cloud.githubusercontent.com/assets/3210225/18217333/1527b62a-7111-11e6-81e2-3912204a110b.png">

**Comment by jcoyne** [comment]
Neither job was currently running, but I didn't see them in the "dead" queue either.  Could this have got lost in a deploy yesterday or due to the disk troubles @cbeer?  The band exhibit only has one item in it. Indexing properties were:
```ruby
metadata: {"last_indexed_estimate"=>11942, "last_indexed_count"=>5240, "last_indexed_finished"=>nil, "last_index_elapsed_time"=>nil, "enqueued_at"=>"2016-08-31 17:10:20 UTC"},
 index_status: 0
```

**Comment by caaster** [comment]
Just restarted Stanford Band reindex.
<img width="1027" alt="screen shot 2016-09-02 at 2 17 26 pm" src="https://cloud.githubusercontent.com/assets/3210225/18218584/7fa22bf0-7118-11e6-9a2b-4dd60c03ce89.png">

**Comment by caaster** [comment]
Just restarted Stanford Stories reindex.
<img width="1019" alt="screen shot 2016-09-02 at 2 22 22 pm" src="https://cloud.githubusercontent.com/assets/3210225/18218615/ca2439de-7118-11e6-9948-c4ce4407196e.png">

**Comment by caaster** [comment]
I am stumped about how to better articulate the problem I am seeing. Here is another screen shot. I don't know if this is an indexing problem or a UI problem. This is one example of a reindexing job that, according to the UI, is still working after several days. Is there something I can do to help characterize the problem more fully?
![screen shot 2016-09-07 at 5 13 49 pm](https://cloud.githubusercontent.com/assets/3210225/18332903/2e4f043e-751f-11e6-8965-9860baf51c95.png)

**Comment by jcoyne** [comment]
It looks like it's still running.  It's possible that it's failed and then restarted itself at least once.

**Comment by jcoyne** [comment]
We've seen several errors due to services we depend on becoming unavailable.  If we broke the reindexing into smaller jobs, it's likely that we could get most of them to pass before one of them trips on a network error, but this is a substantial refactor and requires planning to schedule time to do it.

**Comment by caaster** [comment]
Documenting this morning's status display of the Stanford Band indexing job.
![screen shot 2016-09-08 at 7 16 28 am](https://cloud.githubusercontent.com/assets/3210225/18352702/4cb42efc-7594-11e6-8dab-26e0be74c5ac.png)

**Comment by jcoyne** [comment]
300 items in 14 hours?  It's just tearing through them.

**Comment by jcoyne** [comment]
@caaster one problem affecting indexing performance is https://github.com/sul-dlss/exhibits/issues/355 .  Chris is working on resolving that now.

**Comment by caaster** [comment]
Screenshot updates for 2 indexing jobs.
![screen shot 2016-09-12 at 8 29 45 am](https://cloud.githubusercontent.com/assets/3210225/18441784/29cb9dd0-78c3-11e6-81f8-e60e28b692e1.png)
![screen shot 2016-09-12 at 8 27 49 am](https://cloud.githubusercontent.com/assets/3210225/18441790/2f3940ce-78c3-11e6-86d6-103e9305cc44.png)

**Comment by cbeer** [comment]
It finally got through all the items (in about 24 hours). We fixed a bad configuration on the PURL load-balancer endpoint that caused our error handling behavior in exhibits not to be invoked, decided to have the indexer "power through" all items in the list regardless of what errors it may encounter, and made some general performance improvements for exhibits with really long lists of items.

Additionally, for this particular exhibit, we replaced the references to the 31985 items with a single reference to the collection object, allowing us to take advantage of efficiencies dealing with collection objects.
